### DISCUSSION TITLE
Problem powering my Jetbot

### DISCUSSION THREAD

**Discussion by ozzy1981** [NVIDIA-AI-IOT/jetbot#470] [Q&A]
Hi Guys, so I am at the final stages of assembling the Jetbot but keep coming across an issue. The first problem I am having is that I could not get the battery off Amazon as it is not sold anymore so I thought that I could use a decent power bank instead. This I think is my main problem as the required power for the motors are 3-6V (3 volts @ ~150mA, 6V @ 165mA) whereas the battery I bought ([here](https://www.belkin.com/au/chargers/power-banks/boost-charge-usb-c-pd-power-bank-10k-usb-c-cable/p/p-bpb001/) only puts out 5V (from my understanding). 

I am very new to the world of electronics so please forgive me if this is a bit of a noob question or if this is not the right place to go about asking.

When I plug in both the dev board and the motors the dev board comes up fine as it is plugged into the USB-C port but the featherboard will light up for about 20 seconds, then the green led on the board will go very dim and will not work as you can see in the pictures.
![20210804_192858](https://user-images.githubusercontent.com/19739739/128158035-8aa4afac-63c6-422a-aa3d-bc741887a2f6.jpg)
![20210804_192958](https://user-images.githubusercontent.com/19739739/128158049-43dfa6e2-a4af-4548-a3dc-285df88af158.jpg)

**Comment by santaimpersonator** [comment]
Try the battery linked on the [jetbot.org website](http://jetbot.org/master/bill_of_materials.html) (Nvidia's updated instructions). The [battery listed there](https://amzn.to/2WRcIUe), seems to be in stock.

With that said, I think your battery might be the issue.; you need one that can supply 15W (3A at 5V). Its hard to tell from the specs that Belkin listed, but I think the PD on the USB-C is 18W (2A @ 9V) and USB-A is 12W (2.4A @ 5V). Without an [in-line power meter](https://www.sparkfun.com/products/15571) it'd be hard to tell the exact current limit of the USB-C port at 5V.

**Reply by fifise77** [reply]
Hello I have the same problem, I cannot find the battery liiisted in stock anywhere so I bough the newer model : https://www.amazon.fr/gp/product/B08JTQ66K7/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

but no luck : I have the same behavior than ozzy1981. Does someone have a link to a availed battery that can power successfully a Jetbot ?

**Reply by santaimpersonator** [reply]
### The one that Nvidia recommends (linked above) still seems to be in stock. That's probably your best option.

Otherwise, you can try the [battery we use in our kit](https://www.sparkfun.com/products/15593). However, you will need to follow the [instructions from our hookup guide for the kit](https://learn.sparkfun.com/tutorials/assembly-guide-for-sparkfun-jetbot-ai-kit-v20/3-accessory-installation-to-main-chassis) when using it *(see note below)*.

![image](https://user-images.githubusercontent.com/36016723/132113813-293a6d72-0df5-4ee6-b6ca-6ac8028a79d2.png)

The manufacturer updated the battery pack to support PD/Quick-Charge(?) functionality on both ports, but this caused an issue powering the Jetson Nano. From what we can tell the battery is "too smart".

* The chip inside seems to prioritize one USB port over another.
* When utilizing the port we don't recommend, it appears that the PD/Quick-Charge(?) functionality, for about 1-3ms, the chip checks the prioritized USB port for any PD capabilities or does a negotiation (we suspect, but are not really sure). However, in that 1-3ms time frame, the voltage/current output on the non-recommended USB port becomes limited and this causes the Jetson Nano to brown out.
* We found that the battery doesn't limit the voltage/current on the prioritized USB port, which why we recommend using it.

We tested the problem by hooking up an oscilloscope to the USB power, when connecting the Jetson Nano and we triggered the oscilloscope to capture on an edge. Then we repeated the process connecting a 9V PD/QC device. When comparing the captures, we ended up finding a consistency in the timing with the power failure.

*Based on our experience, you could just have a battery that is "too smart" for it's own good and you just need a dumb battery that does the bare minimum... output a consistent 3A @ 5V.*

**Reply by fifise77** [reply]
thanks for your answer,
I actually found a battery pack that don't have the smart feature and so far seem to work well , except that the 2 USB port a very close so I had to change the USB cable that power the jetson with a strait one

the battery is 
https://www.amazon.fr/gp/product/B07PNL5STG/ref=ppx_yo_dt_b_asin_title_o05_s00?ie=UTF8&psc=1

in case  is can help someone

**Reply by ozzy1981** [reply]
Hey Guys,

Thanks for the responses. I bought the battery from the States (took 2 months to arrive in Australia!!) and unfortunately I am still getting the low power light after about 20 seconds which is now leading me to believe that it is not the battery after all, but could be the motor driver itself? Could I have wired it up wrong?

Thanks in advance for the help

**Reply by santaimpersonator** [reply]
@ozzy1981 Sorry, maybe I misunderstood your issue. Are both the motor driver and Jetson Nano failing due to low power or just the motor driver?

If it is just the motor driver, it could also be that your battery is "too smart" and cutting off power to the motor driver because it isn't drawing enough power (while in standby/not driving the motors). A USB power meter or oscilloscope would be the best to determine what is happening to the voltage output. Otherwise, a simple workaround would be to hookup a separate battery to the motor driver.
* You could try the original battery, just make sure to power it off and on right before you run the motors. So that if there is a shut-off feature, you are able to draw enough power within that 20 sec window to avoid the USB port from shutting down or going into a low power state.
* A simpler option, would be to just hookup a 9V battery to the motor driver. You should be able to find one of these at a local hardware/electronics store for pretty cheap
  ![image](https://user-images.githubusercontent.com/36016723/158424129-6fc0f412-a9c4-4391-8652-9b4f99689101.png)

You could also test if the issue is with the motor driver. In that 20 sec window, try to see if the motor driver responds on the I2C bus and test it again when the LED is dimly lit.
* If the motor driver doesn't respond at all, it is probably the motor driver.
* If it only responds when the LED is powered brightly, then the issue is most likely the power source.
* If it can be pinged on the I2C bus, but doesn't drive the motors when the LED is dimly lit... then the issue is that the voltage/current for driving the motors is too low (i.e. the power source is going into a low-power/standby state).

**Reply by ozzy1981** [reply]
@santaimpersonator all good, thanks for all the advice. To answer your question, I am only having an issue with the motor driver. I have tried connecting 2 batteries to the motor driver and it still goes into a low power state.

The motors work in that 20 second window, then there is no response from the motors at all after it goes into the low power state.

I will try your suggestion with the 9v battery next and see how it goes.

Again, thanks for all the help :)

**Reply by ozzy1981** [reply]
> @ozzy1981 Sorry, maybe I misunderstood your issue. Are both the motor driver and Jetson Nano failing due to low power or just the motor driver?
> 
> If it is just the motor driver, it could also be that your battery is "too smart" and cutting off power to the motor driver because it isn't drawing enough power (while in standby/not driving the motors). A USB power meter or oscilloscope would be the best to determine what is happening to the voltage output. Otherwise, a simple workaround would be to hookup a separate battery to the motor driver.
> 
> * You could try the original battery, just make sure to power it off and on right before you run the motors. So that if there is a shut-off feature, you are able to draw enough power within that 20 sec window to avoid the USB port from shutting down or going into a low power state.
> * A simpler option, would be to just hookup a 9V battery to the motor driver. You should be able to find one of these at a local hardware/electronics store for pretty cheap
>   ![image](https://user-images.githubusercontent.com/36016723/158424129-6fc0f412-a9c4-4391-8652-9b4f99689101.png)
> 
> You could also test if the issue is with the motor driver. In that 20 sec window, try to see if the motor driver responds on the I2C bus and test it again when the LED is dimly lit.
> 
> * If the motor driver doesn't respond at all, it is probably the motor driver.
> * If it only responds when the LED is powered brightly, then the issue is most likely the power source.
> * If it can be pinged on the I2C bus, but doesn't drive the motors when the LED is dimly lit... then the issue is that the voltage/current for driving the motors is too low (i.e. the power source is going into a low-power/standby state).

Update, tried the 9v battery trick and it works!!!
